src/server_manager/web_app/app.ts

```diff
@@ -566,50 +562,45 @@ export class App {
     });
   }
 
-  private showServerIfHealthy(server: server.Server, displayServer: DisplayServer) {
-    server.isHealthy().then((isHealthy) => {
-      if (isHealthy) {
-        // Sync the server display in case it was previously unreachable.
-        this.syncServerToDisplay(server).then(() => {
-          this.showServer(server, displayServer);
-        });
+  private async showServer(server: server.Server, displayServer: DisplayServer): Promise<void> {
+    if (await server.isHealthy()) {
+      // Sync the server display in case it was previously unreachable.
+      await this.syncServerToDisplay(server);
+      await this.showServerManagement(server, displayServer);
+    } else {
+      this.showServerUnreachable(server, displayServer);
```

Comment:
Nit: maybe we add `await` here to be consistent. I notice that you added it to `maybeShowLastDisplayedServer` and `showServerFromRepository` and it seems like a good convention.

I'm also in favor of adding return types to most methods even if they're `async` and return `Promise<void>`, like you did with this method. It could be helpful to add them to the other methods you updated.

Reply:
Done